src/main/java/chessmaster/game/ChessBoard.java

```diff
@@ -205,8 +207,10 @@ public void executeMove(Move move) throws InvalidMoveException {
         getTileAtCoor(startCoor).setTileEmpty(startCoor);
         getTileAtCoor(destCoor).updateTileChessPiece(chessPiece);
 
+
         //@@author onx001
-        if (move.isLeftCastling()) {
+        if (move.isLeftCastling() && (startCoor.getX() - 4)>=0) {
+
             Coordinate rookStartCoor = new Coordinate(startCoor.getX() - 4, startCoor.getY());
```

Comment:
Can consider making use of the `addOffsetToCoordinate` method in the `Coordinate` class to apply offset with boundary checking